File: haskell/005/Main.hs
{-
    ProjectEuler 005
    Haskell / Main.hs
    copyright (c) 2015 Susisu

    GHC 7.8.3
    `ghc Main.hs -o bin/main`
-}

module Main where

main :: IO ()
main = print $ calc 1 20 1

calc :: Int -> Int -> Int -> Int
calc n max acc
    | n > max   = acc
    | otherwise = calc (n + 1) max (acc * (n `div` gcd acc n))

File: julia/005/main.jl
# ProjectEuler 005
# Julia / main.jl
# copyright (c) 2015 Susisu
#
# Julia v0.4.1
# `julia main.jl`

function main()
    p = 1
    for i = 1:20
        p = div(p * i, gcd(p, i))
    end
    println(p)
end

main()
